### DISCUSSION TITLE
Cannot read properties of undefined (reading 'comboDocumentIdentifier')

### DISCUSSION THREAD

**Discussion by krishnagogada** [mkslanc/ace-linters#136] [Q&A]
I integrated Ace Linters using CDN's. 

I got one error from the sentry. 

Error:- Cannot read properties of undefined (reading 'comboDocumentIdentifier')

/ace-linters@latest/build/ace-linters.js in LanguageProvider.$getFileName at line 20826:40
```js
    $getSessionLanguageProvider(session) {
        return this.$sessionLanguageProviders[session["id"]];
    }
    $getFileName(session) {
        let sessionLanguageProvider = this.$getSessionLanguageProvider(session);
        return sessionLanguageProvider.comboDocumentIdentifier;
    }
  ```
/ace-linters@latest/build/ace-linters.js in LanguageProvider.doHover at line 21031:46
/ace-linters@latest/build/ace-linters.js in HoverTooltip.$gatherData at line 20972:18
/ace-linters@latest/build/ace-linters.js in HoverTooltip.waitForHover at line 19639:18

I try to understand the code. To my understanding to get `sessionLanguageProviders` we have to set `sessionLanguageProviders` with the `session["id"]`.

But from the below code, it is like registering the editor and then registering the session.

language-provider.ts
![image](https://github.com/user-attachments/assets/14fb347e-e4d0-4bbe-8f63-1620436e533b)

But it should be reversed. 1st register the session and then register the editor.

I am not sure about this. But I just had this doubt. Or else is there any possibility of making a mistake from the integration side while integrating through CDN?

Do you guys have any documentation regarding the functionality options? I have this doubt about what `hover: boolean` will do in functionality?

**Comment by mkslanc** [comment]
@krishnagogada hey! You could look to [that example](https://github.com/ajaxorg/ace-samples/tree/master/samples/ace-builds-cdn) or [sandbox example](https://mkslanc.github.io/ace-playground/#validation-against-json-schema). They both are using cdn integration. 

As for `hover` you could look to [declaration file with all options/properties](https://github.com/mkslanc/ace-linters/blob/main/packages/ace-linters/types/types/language-service.d.ts#L170), but it simply just turning on/off exact feature from all services; so instead of turning off in each service, you could just set it in provider and no `hover` calls would be used.

**Reply by krishnagogada** [reply]
@mkslanc 

1. Checked the given examples of CDN integration, my code is also very similar to that. So is that a bug?
2. Yes I can see the declaration file, but there is no information about the hover feature. This means if hover is **ON** then what changes can see in the UI, similarly if hover is **OFF** what kind of changes we can see in the UI.

**Reply by mkslanc** [reply]
1. I think I need to look at your code, could you share minimal snippet of not working code?
2. It would disable hover hint about token, when you are positioning your cursor on it. For example this one: 
![image](https://github.com/user-attachments/assets/2b203b6f-e5df-442d-a53e-e91ac0a3db6e)
